spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java

```diff
@@ -264,6 +279,27 @@ public boolean isAcquiredInThisProcess() {
 			return this.mutex.isAcquired(this.path);
 		}
 
+		public boolean renew() {
+			if (!this.delegate.isHeldByCurrentThread()) {
+				throw new IllegalMonitorStateException("You do not own mutex at " + this.path);
+			}
+			while (true) {
+				try {
+					boolean renewed = this.mutex.renew(this.path);
+					if (renewed) {
+						this.lastUsed = System.currentTimeMillis();
+					}
+					return renewed;
+				}
+				catch (TransientDataAccessException e) {
```

Comment:
I think the `TransactionTimedOutException` has to be caught over here as well.
See `tryLock()`  for example.

Reply:
Is `TransactionTimedOutException` thrown with default transaction timeout too ? You should be right.

Reply:
Correct, it is thrown even with default timeout, which is not infinite.